zerver/views/user_settings.py

```diff
@@ -96,13 +96,14 @@ def confirm_email_change(request: HttpRequest, confirmation_key: str) -> HttpRes
         context=context,
         realm=user_profile.realm,
     )
-
+    new_email_list = escape(new_email).split("@")
+    old_email_list = escape(old_email).split("@")
```

Comment:
We should use the `Address` module for this, rather than parsing manually.

Reply:
@timabbott does it mean that I should write a function in `addressee.py` and use that function to parse?

Reply:
You should use that function rather than splitting on `@`, which I think may be wrong in some strange corner cases. It's part of the Python 3 standard library -- you can read about it and use it correctly, or just grep for examples in our codebase to see how it's used.